src/backend/api/history_endpoints.py

```
@@ -57,18 +57,30 @@ def list_history(
     offset = max(0, offset)
     q = q.strip() if q else None
 
+    # group: a run_groups id (uuid) or the literal "ungrouped". Groups are
+    # personal, so the filter needs no extra authorization: rows are already
+    # scoped below, and filtering your view by someone else's uuid just
+    # yields rows you could see anyway that happen to sit in that group.
+    if group and group != "ungrouped":
+        try:
+            group = str(uuid.UUID(group))
+        except ValueError:
+            raise HTTPException(status_code=400, detail="Invalid group id")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)